src/python/pants/backend/jvm/tasks/coursier_resolve.py

```diff
@@ -91,7 +86,7 @@ def _compute_jars_to_resolve_and_pin(raw_jars, artifact_set, manager):
 
     return jar_list, untouched_pinned_artifact
 
-  def resolve(self, targets, compile_classpath):
+  def resolve(self, targets, compile_classpath, sources, javadoc):
```

Comment:
How would you feel about making the new params optional with default values of False?

Reply:
Good question. I thought about that and the settled with this idea:
I think generally it is a good practice especially when the call stack is relatively simple. However in cases where the same args get passed further down into other functions, things may look confusing, so making it non-optional would make it clear during debugging where the args are came from, and also more clear for readers that just want to skim the code.

For example:
```
def resolve(..., sources=false, javadoc=false)
  x(sources, javadoc)

def x(sources,javadoc)
  y(sources,javadoc)

def y(sources, javadoc):
  // do other things
```

basically as a reader, it is hard to tell if `sources`, `javadoc` in `y` actually comes from the default value in `resolve` or the caller of `resolve` unless going back on the stack and then check how `resolve` is getting called, whereas non-optional args would make it clear that `sources`, `javadoc` in `y` is the exactly how `resolve` is getting called.

Reply:
sounds reasonable to me.

Reply:
The other approach that resolve methods take is to ask for a list of confs.